test/Polly.Core.Tests/CircuitBreaker/BrokenCircuitExceptionTests.cs

```diff
@@ -7,10 +7,27 @@ public class BrokenCircuitExceptionTests
     [Fact]
     public void Ctor_Ok()
     {
-        new BrokenCircuitException().Message.Should().Be("The circuit is now open and is not allowing calls.");
-        new BrokenCircuitException("Dummy.").Message.Should().Be("Dummy.");
-        new BrokenCircuitException("Dummy.", new InvalidOperationException()).Message.Should().Be("Dummy.");
-        new BrokenCircuitException("Dummy.", new InvalidOperationException()).InnerException.Should().BeOfType<InvalidOperationException>();
+        var retryAfter = new TimeSpan(1, 0, 0);
+        var defaultException = new BrokenCircuitException();
+        defaultException.Message.Should().Be("The circuit is now open and is not allowing calls.");
+        defaultException.RetryAfter.Should().BeNull();
+        var retryAfterException = new BrokenCircuitException(retryAfter);
+        retryAfterException.Message.Should().Be($"The circuit is now open and is not allowing calls. It can be retried after '{retryAfter}'.");
+        retryAfterException.RetryAfter.Should().Be(retryAfter);
+        var dummyMessageException = new BrokenCircuitException("Dummy.");
+        dummyMessageException.Message.Should().Be("Dummy.");
+        dummyMessageException.RetryAfter.Should().BeNull();
+        var dummyMessageWithRetryAfterException = new BrokenCircuitException("Dummy.", retryAfter);
+        dummyMessageWithRetryAfterException.Message.Should().Be("Dummy.");
+        dummyMessageWithRetryAfterException.RetryAfter.Should().Be(retryAfter);
+        var dummyMessageExceptionWithInnerException = new BrokenCircuitException("Dummy.", new InvalidOperationException());
+        dummyMessageExceptionWithInnerException.Message.Should().Be("Dummy.");
+        dummyMessageExceptionWithInnerException.InnerException.Should().BeOfType<InvalidOperationException>();
+        dummyMessageExceptionWithInnerException.RetryAfter.Should().BeNull();
+        var dummyMessageExceptionWithInnerExceptionAndRetryAfter = new BrokenCircuitException("Dummy.", retryAfter, new InvalidOperationException());
+        dummyMessageExceptionWithInnerExceptionAndRetryAfter.Message.Should().Be("Dummy.");
+        dummyMessageExceptionWithInnerExceptionAndRetryAfter.InnerException.Should().BeOfType<InvalidOperationException>();
+        dummyMessageExceptionWithInnerExceptionAndRetryAfter.RetryAfter.Should().Be(retryAfter);
```

Comment:
For readability:

```suggestion
        var retryAfter = new TimeSpan(1, 0, 0);

        var defaultException = new BrokenCircuitException();
        defaultException.Message.Should().Be("The circuit is now open and is not allowing calls.");
        defaultException.RetryAfter.Should().BeNull();

        var retryAfterException = new BrokenCircuitException(retryAfter);
        retryAfterException.Message.Should().Be($"The circuit is now open and is not allowing calls. It can be retried after '{retryAfter}'.");
        retryAfterException.RetryAfter.Should().Be(retryAfter);

        var dummyMessageException = new BrokenCircuitException("Dummy.");
        dummyMessageException.Message.Should().Be("Dummy.");
        dummyMessageException.RetryAfter.Should().BeNull();

        var dummyMessageWithRetryAfterException = new BrokenCircuitException("Dummy.", retryAfter);
        dummyMessageWithRetryAfterException.Message.Should().Be("Dummy.");
        dummyMessageWithRetryAfterException.RetryAfter.Should().Be(retryAfter);

        var dummyMessageExceptionWithInnerException = new BrokenCircuitException("Dummy.", new InvalidOperationException());
        dummyMessageExceptionWithInnerException.Message.Should().Be("Dummy.");
        dummyMessageExceptionWithInnerException.InnerException.Should().BeOfType<InvalidOperationException>();
        dummyMessageExceptionWithInnerException.RetryAfter.Should().BeNull();

        var dummyMessageExceptionWithInnerExceptionAndRetryAfter = new BrokenCircuitException("Dummy.", retryAfter, new InvalidOperationException());
        dummyMessageExceptionWithInnerExceptionAndRetryAfter.Message.Should().Be("Dummy.");
        dummyMessageExceptionWithInnerExceptionAndRetryAfter.InnerException.Should().BeOfType<InvalidOperationException>();
        dummyMessageExceptionWithInnerExceptionAndRetryAfter.RetryAfter.Should().Be(retryAfter);
```

Reply:
Or you could create parameterized test instead. Something like this
```cs
[Theory]
[InlineData(null, false, "The circuit is now open and is not allowing calls.")]
[InlineData(null, true, "The circuit is now open and is not allowing calls. It can be retried after.")]
[InlineData("Dummy.", false, "Dummy.")]
[InlineData("Dummy.", true, "Dummy.")]
public void bce_without_inner_exception(string? errorMessage, bool useRetryAfter, string expectedMessage)
{
    //Arrange
    TimeSpan? retryAfter = useRetryAfter ? new(1, 0, 0) : null;
    List<object> args = new();
    if (errorMessage is not null) { args.Add(errorMessage); }
    if (useRetryAfter) { args.Add(retryAfter); }
    
    //Act
    var exception = (BrokenCircuitException)Activator.CreateInstance(typeof(BrokenCircuitException), args.ToArray());
    
    //Assert
    exception.Message.Should().Be(expectedMessage);
    exception.RetryAfter.Should().Be(retryAfter);
}
```
The same could be applied for those cases where you pass an inner `Exception`

Reply:
IMHO there's too much conditional logic in that example.

Either it's one test broken up, or multiple tests (one for each constructor).

Reply:
I may opt for separating the tests across multiple methods to make things more manageable.